python/ray/node.py

```diff
@@ -390,7 +392,9 @@ def plasma_store_socket_name(self):
     @property
     def unique_id(self):
         """Get a unique identifier for this node."""
-        return f"{self.node_ip_address}:{self._plasma_store_socket_name}"
+        if hasattr(self, "_plasma_store_socket_name"):
```

Comment:
We still have things like self.raylet_ip_address when no_worker is True, which I think is not correct?